### DISCUSSION TITLE
Setting color to 255,0,225 ends up setting the color to red

### DISCUSSION THREAD

**Discussion by AidenJames1995** [jath03/openrgb-python#37] [Q&A]
I'm just getting started with OpenRGB and OpenRGB-Python and I'm having problems getting something as simple as setting a color programmatically to stick.

Here's my code:
`from openrgb import OpenRGBClient
from openrgb.utils import RGBColor, DeviceType

client = OpenRGBClient()

client.clear() # Turns everything off
motherboard = client.get_devices_by_type(DeviceType.MOTHERBOARD)[0]
motherboard.set_color(RGBColor(255,0,255))
motherboard.set_mode("static")
`

Admittingly, I'm not the best with python, I'm more a PHP programmer, but I don't see anything wrong with the code here. When I run this, the motherboard and the devices plugged into the addressable header on the board will flash purple for about 0.25 second and then fall back to red. If I manually set a color in OpenRGB then run this, it will set it to the color selected in OpenRGB after turning everything off for a brief moment. For whatever reason it seems to be tied to set_mode. When ever I run this without that call it seems to works fine.

**Comment by AidenJames1995** [comment]
Well I think I fixed that, apparently my motherboard only likes direct mode. Now I'm trying to get my RAM to set a color, but nothing:
Code now:
```
#!/usr/bin/env python3
from openrgb import OpenRGBClient
from openrgb.utils import RGBColor, DeviceType

client = OpenRGBClient()
client.clear() # Turns everything off
motherboard = client.get_devices_by_type(DeviceType.MOTHERBOARD)[0]
motherboard.set_mode("direct")
motherboard.set_color(RGBColor(255, 0, 255))

dram = client.get_devices_by_type(DeviceType.DRAM)
for module in range(len(dram)):
    ram = client.get_devices_by_type(DeviceType.DRAM)[module]
    ram.set_mode("direct")
    ram.set_color(RGBColor(255, 0, 255))

```
RAM sticks are G.Skill TridentZ Royal
Motherboard is Asus Prime X570-Pro

**Reply by AidenJames1995** [reply]
Needed to time.sleep(.05). Final Code:

```
#!/usr/bin/env python3
from openrgb import OpenRGBClient
from openrgb.utils import RGBColor, DeviceType
import time

client = OpenRGBClient()
client.clear() # Turns everything off

motherboard = client.get_devices_by_type(DeviceType.MOTHERBOARD)
dram = client.get_devices_by_type(DeviceType.DRAM)
devices = motherboard + dram
for device in range(len(devices)):
    time.sleep(.05)
    d = client.devices[device]
    d.set_mode("direct")
    d.set_color(RGBColor(255, 0, 255))

```

**Reply by jath03** [reply]
> Well I think I fixed that, apparently my motherboard only likes direct mode.

I'd guess that this isn't true and that it's because you're probably trying to use `set_color` set the colors in a mode that is not a per-led mode (non per-led meaning that you can only set two colors for an effect or whatever when you can change like 30 leds in direct mode).  I am going to add the ability to use `set_color`/`set_colors` for that sometime soon, but for now, you will have to modify the mode's colors manually.  That would look something like this:
```python
static = devices.modes[0] # assuming that static is the first mode in the list
static.colors[0] = RGBColor(255, 0, 255)
device.set_mode(static)

# or

device.modes[0].colors[0] = RGBColor(255, 0, 255)
device.set_mode('static')
```
